### PR TITLE
output/eve: add 'verdict' field to 'alert' and 'drop' events - v4

### PR DESCRIPTION
**Pull request by jufajardini**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5464

Previous PR: https://github.com/OISF/suricata/pull/8348

Changes from last PR:
- add `optional` to the `verdict` field in schema.json
- update copyright years

suricata-verify-pr: 1124
https://github.com/OISF/suricata-verify/pull/1124

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .tcp.rst | 145757 | 113528 | 77.89% |
| | SURI_TLPR1_stats_chk | | |
| .flow.memuse | 905993032 | 521782368 | 57.59% |
| | TREX_GENERIC_stats_chk | | |
| .tcp.segment_from_pool | 14826199 | 15609962 | 105.29% | 

Pipeline 12404

File: src/output-json-alert.c [hunk 1]
```diff
@@ -828,12 +844,16 @@ static int AlertJson(ThreadVars *tv, JsonAlertLogThread *aft, const Packet *p)
             jb_set_string(jb, "capture_file", pcap_filename);
         }
 
+        if (PacketActionVerdictToString(p) != NULL) {
```
Comment (jlucovsky) [inline src/output-json-alert.c:847]:
```suggestion
        const char *verdict = PacketActionVerdictToString(p);
        if (vedict != NULL) {
            jb_set_string(jb, "verdict", verdict);
        }
```

File: src/output-json-alert.c [hunk 2]
```diff
@@ -1059,6 +1079,16 @@ static void JsonAlertLogSetupMetadata(AlertJsonOutputCtx *json_output_ctx,
...
+    /* action is enabled by default */
+    flags |= LOG_JSON_ACTION;
+    const char *action = ConfNodeLookupChildValue(conf, "action");
+    if (action != NULL) {
+        if (ConfValIsFalse(action)) {
+            SetFlag(conf, "action", LOG_JSON_ACTION, &flags);
+            g_logalert_action = false;
+        }
+    }
+
```
Comment (jlucovsky) [inline src/output-json-alert.c:1091]:
```suggestion
    /* action is enabled by default */
    flags |= LOG_JSON_ACTION;
    SetFlag(conf, "action", LOG_JSON_ACTION, &flags);
    if (!(flags & LOG_JSON_ACTION)) {
            g_logalert_action = false;
    }
```

File: src/output-json-alert.c (hunk 1 above)
Comment (jlucovsky) [reply src/output-json-alert.c:847]: This avoids calling `PacketCheckAction` more than is necessary.

File: src/output-json-alert.c (hunk 2 above)
Comment (jlucovsky) [reply src/output-json-alert.c:1091]: This avoids calling `ConfNodeLookupChildValue` twice.

**Review by jlucovsky — COMMENTED** [review]
Looks good -- minor changes suggested.

**Comment by jufajardini** [comment]
Feedback incorporated in: https://github.com/OISF/suricata/pull/8558

**Pull request closed without merging**